frontend/src/Editor/CodeBuilder/CodeHinter.jsx

```diff
@@ -116,30 +120,124 @@ export function CodeHinter({
+
+          resolve();
+        }
+      }
+    }).then(() => {
+      setIsOpen(true);
+      forceUpdate();
+    });
+  };
+  const [, forceUpdate] = React.useReducer((x) => x + 1, 0);
+
   return (
-    <div style={{ width: '100%' }}>
+    <div className="code-hinter-wrapper" style={{ width: '100%' }}>
+      {usePortalEditor && <CodeHinter.PopupIcon callback={handleToggle} />}
       <div
         className={`code-hinter ${className || 'codehinter-default-input'}`}
         key={suggestions.length}
         style={{ height: height || 'auto', minHeight, maxHeight: '320px', overflow: 'auto' }}
       >
-        <CodeMirror
-          value={initialValue}
-          realState={realState}
-          scrollbarStyle={null}
+        <CodeHinter.UsePortal
+          isOpen={isOpen}
+          callback={setIsOpen}
+          componentName={componentName}
+          key={suggestions.length}
+          theme={theme}
+          getPreview={enablePreview && getPreview}
           height={height}
-          onFocus={() => setFocused(true)}
-          onBlur={(editor) => {
-            const value = editor.getValue();
-            onChange(value);
-            setFocused(false);
-          }}
-          onChange={(editor) => valueChanged(editor, onChange, suggestions, ignoreBraces)}
-          onBeforeChange={(editor, change) => onBeforeChange(editor, change, ignoreBraces)}
-          options={options}
-        />
+          forceUpdate={forceUpdate}
+        >
+          <CodeMirror
+            value={initialValue}
+            realState={realState}
+            scrollbarStyle={null}
+            height={height || 'auto'}
+            onFocus={() => setFocused(true)}
+            onBlur={(editor) => {
+              const value = editor.getValue();
+              onChange(value);
+              setFocused(false);
+            }}
+            onChange={(editor) => valueChanged(editor, onChange, suggestions, ignoreBraces)}
+            onBeforeChange={(editor, change) => onBeforeChange(editor, change, ignoreBraces)}
+            options={options}
+            viewportMargin={Infinity}
+          />
+        </CodeHinter.UsePortal>
       </div>
-      {enablePreview && getPreview()}
+
+      {enablePreview && !isOpen && getPreview()}
     </div>
   );
 }
+
+const PopupIcon = ({ callback }) => {
+  return (
+    <div className="d-flex justify-content-end">
+      <OverlayTrigger
+        trigger={['hover', 'focus']}
+        placement="top"
+        delay={{ show: 800, hide: 100 }}
+        overlay={<Tooltip id="button-tooltip">{'Pop out code editor into a new window'}</Tooltip>}
+      >
+        <img
+          className="svg-icon m-2 popup-btn"
+          src="/assets/images/icons/portal-open.svg"
+          width="12"
+          height="12"
+          onClick={(e) => {
+            e.stopPropagation();
+            callback();
+          }}
+        />
+      </OverlayTrigger>
+    </div>
+  );
+};
+
+const UsePortal = ({ children, ...restProps }) => {
```

Comment:
I think this should go under `_hooks` folder because we can reuse it for popover and modal also later

Reply:
yeah could do that